integration/test/test_progress.py

```diff
@@ -0,0 +1,379 @@
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+import sys
+import unittest
+import os
+import numpy
+import socket
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from integration.test import geopm_context
+from integration.test import util
+import geopmpy.io
+import geopmpy.hash
+from integration.experiment import machine
+import geopm_test_launcher
+
+
+class AppConf(object):
+    """Class that is used by the test launcher in place of a
+    geopmpy.io.BenchConf when running the progress benchmark.
+
+    """
+    def write(self):
+        """Called by the test launcher prior to executing the test application
+        to write any files required by the application.
+
+        """
+        pass
+
+    def get_exec_path(self):
+        """Path to benchmark filled in by template automatically.
+
+        """
+        return util.get_exec_path('test_progress')
+
+    def get_exec_args(self):
+        """Returns a list of strings representing the command line arguments
+        to pass to the test-application for the next run.  This is
+        especially useful for tests that execute the test-application
+        multiple times.
+
+        """
+        return []
+
+
+class TestIntegration_progress(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Create launcher, execute benchmark and set up class variables.
+
+        """
+        sys.stdout.write('(' + os.path.basename(__file__).split('.')[0] +
+                         '.' + cls.__name__ + ') ...')
+        cls._test_name = 'test_progress'
+        cls._report_path = '{}.report'.format(cls._test_name)
+        cls._trace_path = '{}.trace'.format(cls._test_name)
+        cls._agent_conf_path = cls._test_name + '-agent-config.json'
+        # Clear out exception record for python 2 support
+        geopmpy.error.exc_clear()
+
+        # Create machine
+        cls._machine = machine.Machine()
+        try:
+            cls._machine.load()
+            sys.stderr.write('Warning: {}: using existing file "machine.json", delete if invalid\n'.format(cls._test_name))
+        except RuntimeError:
+            cls._machine.save()
+
+        # Set the job size parameters
+        cls._num_node = 1
+        cls._num_rank = 10
+        cls._num_active_cpu = cls._machine.num_core() - 2 * cls._machine.num_package()
```

Comment:
Speaking of the Machine object, it would be great if we could ask it for the number of active CPUs to avoid sprinkling the magic ```2``` around all the tests.  This will be nice if/when we change where and how the Controller runs (application mode).

Reply:
The unit test has been failing since we started deriving the hint from the ApplicationStatus lock free solution and not based on messages in the ApplicationRecordLog.  This unit test needs to be updated to mock the ApplicationStatus object, not the ApplicationRecordLog object.

In terms of the magic number 2, this is based on the geopmpy.launch behavior (reserve one core for the OS and one core for GEOPM, which is what this number two reprsents).  The machine file does not contain information about the launch command was issued, or what the application cpu affinities are.